go-controller/pkg/ovn/policy.go

```diff
@@ -744,11 +758,9 @@ func (oc *Controller) addNetworkPolicyPortGroup(policy *knet.NetworkPolicy) {
 			ingress.addPortPolicy(&portJSON)
 		}
 
-		hashedLocalAddressSet := ""
-		// peerPodAddressMap represents the IP addresses of all the peer pods
-		// for this ingress.
-		peerPodAddressMap := make(map[string]bool)
-		if len(ingressJSON.From) != 0 {
+		var hashedLocalAddressSet string
+		var peerPodAddressMap map[string]bool
```

Comment:
I don't like changing this to make this nil unless `if hasAnyLabelSelector` but then code below assumes it's non-nil based on a slightly-different (though in fact equivalent) check... I feel like it's better to just initialize it unconditionally

Reply:
@danwinship  we were unnecessarily caling `make(map[string]bool)` and we weren't using `peerPodAddressMap` anywhere else in no `labelselector` case. Do you want me to do this

```
hashedLocalAddressSet := ""
peerPodAddressMap := nil
```
or just

```
hashedLocalAddressSet := ""
```

Reply:
the code is currently doing

    var peerPodAddressMap map[string]bool
    if X {
            peerPodAddressMap = make(map[string]bool)
    }
    if Y {
            peerPodAddressMap[foo] = bar
    }

It works because currently Y can only be true if X was true, but if the conditions changed slightly then you could end up inside the second `if` body with a `nil` map and then panic. I feel like that's bad coding style and it would be better to initialize the map even when it wasn't going to be used. Maybe other people don't agree with that.

Reply:
@danwinship I have addressed your comment above. Can you PTAL?